### PR TITLE
[SMALLFIX] Extract create UfsBaseFileSystem for extension

### PR DESCRIPTION
**Pull request by jiacheliu3**
### What changes are proposed in this pull request?

Extract the logic to create `UfsBaseFileSystem` into `FileSystemContext`, for possible extension. This change is functionally a refactor that changes nothing.

### CODE REVIEW

File: dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java [hunk 1]
```diff
@@ -1001,6 +1003,18 @@ private void initializeLocalWorker() throws IOException {
     mLocalWorkerInitialized = true;
   }
 
+  /**
+   * Creates an underlying file system which handles UFS fallback.
+   *
+   * @param ufsOptions options to access the UFS
+   * @return a UFS-based FileSystem implementation
+   */
+  public FileSystem createUfsBaseFileSystem(Optional<UfsFileSystemOptions> ufsOptions) {
```
Comment (dbw9580) [inline dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java:1012]: I think you can just accept the argument without the `Optional` part, and let the callers to make sure they do have a `UfsFileSystemOptions`.

**Review by dbw9580 — APPROVED** [review]
LGTM with a minor comment

File: dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java (hunk 1 above)
Comment (dbw9580) [reply dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java:1012]: The callers typically have more context in handling an empty `Optional<UfsFileSystemOptions>` than `FileSystemContext`, e.g. using a default, or throwing an error to alert the user.

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**